### ISSUE TITLE
Support for setting ANSI SQL mode

### ISSUE THREAD

**Issue by tiksn** [dolthub/dolt#4663]
I am still trying to use [XCA](https://hohnstaedt.de/xca/) for storing certificates in Dolt repository.

With a dolt version `0.50.10` and database storage format `NEW ( __DOLT__ )` I can't do that because I am getting error

Query:

```SQL
SET SESSION SQL_MODE='ANSI'
```

Error Message:
> value ANSI was not found in the set

[MySQL Documentation reference](https://dev.mysql.com/doc/refman/8.0/en/sql-mode.html#sqlmode_ansi)

Previous issue fix #4436 helped me to start working with XCA, but after closing and opening the app data is gone (not really, it's just cane read because of this issue, if I am trying `dolt diff` I can see that data is there).

**Comment by timsehn** [comment]
We'll get started on this one Monday. Should be an easy fix.  Thanks for being patient with us.

**Comment by fulghum** [comment]
Heya @tiksn, sorry for the delay on this one. We're excited to hear if you can get XCA working with Dolt! 🤞 

I'm digging into this one and will get a PR out shortly. Stay tuned...

**Comment by fulghum** [comment]
We got the fix for this merged into main for `go-mysql-server` and are building it into Dolt now. Apologies again that we didn't get to this one earlier in the week. I'll update again once we get the fix released in the next Dolt version.

**Comment by fulghum** [comment]
This is now released in [Dolt version 0.50.14](https://github.com/dolthub/dolt/releases/tag/v0.50.14) 🎉 This version unblocks you from setting `sql_mode` to `ANSI`, but note that Dolt doesn't yet honor all the settings for ANSI mode. Depending on how much of ANSI that XCA's queries rely on, we could hit more snags, but don't hesitate to cut us new issues and we'll be happy to dig in and see what we can do.  

Thanks again for taking the time to report this. Let us know how the next round of testing goes! 🤞
